client/operation_read_object_group.go

```diff
@@ -19,6 +19,48 @@ func (l appLogger) Log(args ...interface{}) {
 	log.Printf("AWS: %+v", args...)
 }
 
+// processFilterJSON takes a JSON string as input, checks the structure of the "regex" key,
+// and returns a modified JSON string according to the specified rules.
+// For example:
+//     `{"AND":[{"field":"key","regex":{"pattern":".*","strict":true}}]}`
+// is converted back into
+//     `{"AND":[{"field":"key","regex":".*"}]}`
+func processFilterJSON(input string) (string, error) {
```

Comment:
Unclear interface. Function name is too generic and gives no hints to what it's doing.

Perhaps something like this:
```
// BUG: ChaosSearch breaking API change now uses two representations of "regex", one as an object, the other as a string
// regexObject2String takes a filterJson string and converts any regex object to a regex string
func regexObject2String(filterJson string) (string, error) {
```

Reply:
Good point with improving the comment!

However, since the variable, that this function processes, is called **FilterJSON**, the proposed name **regexObject2String** reads too generic for my taste.